ckanext/archiver/reports.py

```diff
@@ -175,10 +175,14 @@ def broken_links_for_organization(organization, include_sub_organizations=False)
         elif er.startswith("DATA4NR"):
             via = "Data4nr"
 
-        archived_resource = model.Session.query(model.ResourceRevision)\
-                                 .filter_by(id=resource.id)\
-                                 .filter_by(revision_timestamp=archival.resource_timestamp)\
-                                 .first() or resource
+        try:
+            archived_resource = model.Session.query(model.ResourceRevision)\
+                .filter_by(id=resource.id)\
+                .filter_by(revision_timestamp=archival.resource_timestamp)\
+                .first() or resource
+        except AttributeError:
+            # CKAN 2.9 doesn't have revisions, TODO check if activity stream can help
+            archived_resource = resource
```

Comment:
It's not exactly broken. It just won't provide the specific feature of "this resource used to have invalid URL X but it is now updated to Y". Instead, it will always work with the latest URL for each resource. It's still a meaningful report.

I'll take a look at what the activity stream includes, though.

Reply:
Looks like it would be possible to recreate this functionality from the activity stream, but with significantly increased overhead.

Either we would need to link the `activity` table to `activity_details` and search for entries with a matching "activity" timestamp and with "activity_detail" representing an update on the relevant resource, or we would need to perform a search on the `activity_detail` table using only the resource ID, which is not an indexed field. The first option will likely be faster, but both are likely to be slower than the old implementation.

Reply:
Activity stream has an implementation, but I'm not sure how well it will work, or whether it's really a significant feature anyway.